hub/core/meta/tensor_meta.py

```diff
@@ -50,16 +52,41 @@ def create(
     def load(key: str, storage: StorageProvider):
         return TensorMeta(get_tensor_meta_key(key), storage)
 
-    def update_tensor_meta_with_array(self, array: np.ndarray, batched=False):
+    def check_batch_is_compatible(self, array: np.ndarray):
+        # TODO: docstring (note that `array` is assumed to be batched)
+
+        # dtype is only strict after a sample exists
+        if self.dtype != array.dtype.name:
+            raise TensorMetaMismatchError("dtype", self.dtype, array.dtype.name)
+
+        sample_shape = array.shape[1:]
+
+        if self.length > 0:
```

Comment:
Why not `len(self)`?

Reply:
because `TensorMeta` has no length, it only has a length property. and given that `tensor_meta.json` will be used in javascript also, the `__len__` naming convention wouldn't make much sense there.

`len(TensorMeta)` tells me that the meta itself is what the length is. this is just ambiguous to me.

maybe `length` should instead be `sample_count`

Reply:
but i will keep `length` for now

Reply:
OH I see.